tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+        else:
+            print('Detail type not recognized: {}'.format(info_type))
+
+        detail = self._get_detail()
+        return Info(info_type, verb, task, extra, date, time, detail, line)
+
+    def _get_result(self, line):
+        parts = line.split(' ')
+        if len(parts) < 3:
+            print(parts)
+            return None
+        date = parts[0]
+        time = parts[1]
+        result_type = parts[2]
+        level = parts[3].split(':')[0]
+        number = parts[-1]
+
+        if result_type == FAILED_TYPE:
+            detail = self._get_detail(other_limit=-1)
+        else:
+            detail = None
+
+        return Result(result_type, level, number.strip(), date, time, detail, line)
+
+    def _get_action(self, line):
+        parts = line.split(' ')
+        if len(parts) < 3:
+            return None
+        date = parts[0]
+        time = parts[1]
+        verb = parts[2]
+        task = parts[3]
+        return Action(verb, task.split('...')[0], date, time, line)
+
+    def _get_operation(self, line):
+        parts = line.split(' ')
+        if len(parts) < 3:
+            return None
+        date = parts[0]
+        time = parts[1]
+        verb = parts[2]
+        task = None
+        extra = ' '.join(parts[3:])
+        return Operation(verb, task, extra, date, time, line)
+
+
+def _make_parser():
+    # type: () -> argparse.ArgumentParser
+    parser = argparse.ArgumentParser(
+        description="""
+Analyze the spread log and generates a file with a standarized output. Il also
+allows to filter the output by type and define the number of lines to show
+for the error/debug/warning output.
+"""
+    )
+    parser.add_argument(
+        "-c",
+        "--cut",
+        type=int,
+        default=1000,
+        help="maximun number of lines for logs on errors and debug sections",
+    )
+    parser.add_argument(
+        "-f",
+        "--format",
+        type=str,
+        default="json",
+        choices=['json'],
+        help="format for the output",
+    )
+    parser.add_argument(
+        "-pd",
+        "--print-details",
+        type=str,
+        default=NONE,
+        choices=[ALL, ERROR, OPERATION, ACTION, INFO, NONE],
+        help="Filter which info to print",
+    )
+    parser.add_argument(
+        "-pr",
+        "--print-results",
+        type=str,
+        default=NONE,
+        choices=[ALL, FAILED, ABORTED, SUCCESSFUL, NONE],
+        help="Filter which results to print",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        default="spread-results.json",
+        type=str,
+        help="output file to save the result",
+    )
+    parser.add_argument(
+        "--store-setup",
+        action="store_true",
+        help="will save all the text before the spread run is started",
+    )
+    parser.add_argument(
+        "logpath", metavar="PATH", help="path to the log to be analyzed"
+    )
+    return parser
+
+
+def main():
+    # type: () -> None
```

Comment:
It would be nice to add type annotations for the rest of the file, where it may find mistakes.